distributed/stealing.py

```diff
@@ -234,14 +242,13 @@ def steal_time_ratio(self, ts: TaskState) -> tuple[float, int] | tuple[None, Non
         if not ts.dependencies:  # no dependencies fast path
             return 0, 0
 
-        assert ts.processing_on
-        ws = ts.processing_on
-        compute_time = ws.processing[ts]
+        compute_time = self.scheduler.get_task_duration(ts)
 
         if not compute_time:
             # occupancy/ws.proccessing[ts] is only allowed to be zero for
             # long running tasks which cannot be stolen
-            assert ts in ws.long_running
+            assert ts.processing_on
+            assert ts in ts.processing_on.long_running
```

Comment:
Codecov says this is not hit but I'd expect this to be hit by `test_dont_steal_long_running_tasks`

We should probably look at that test, maybe some test assumption is no longer true

Reply:
Ok, `test_dont_steal_long_running_tasks` is pretty meaningless. I believe something went wrong during a refactoring of that test. It only asserts that long_tasks are only "executed" once, i.e. they are not rescheduled. This is entirely irrelevant for stealing.

Reply:
Given that this test was already this way I suggest to set this out of scope for this PR. I might be able to address this with https://github.com/dask/distributed/pull/7030